doc/md/predicates.md

```diff
@@ -89,6 +89,62 @@ client.Pet.
 
 Custom predicates can be useful if you want to write your own dialect-specific logic or to control the executed queries.
 
+For example, if you want to use builtin sql functions such as a `DATE()` function to a `WHERE` statement.
+It can be achieved by using either `P()` or `ExprP()`.
+
+1. `P()`
+
+Using `P()` gives more flexibility in how you arrange the strings.
+
+```go
+Select("id").
+From(Table("users")).
+Where(P(func(b *Builder) {
+    b.WriteString("DATE(").Ident("last_login_at").WriteString(") >= ").Arg("2022-05-03")
+}))
+```
+
+The above code will produce the following SQL query:
+
+```sql
+SELECT `id` FROM `users` WHERE DATE(`last_login_at`) >= ?
+```
+
+2. `ExprP()`
+
+The same thing can be achieved with
```

Comment:
```suggestion
2\. Invalid a predicate expression using the `ExprP()` option:
```

Reply:
Hi @a8m, thank you for all the feedback. I have made the changes as you suggested.

However for this, I do not understand the wording ` Invalid a predicate expression...`. Are we invalidating anything?

Reply:
Sorry for the confusion. I think I meant “inline”, but accidentally wrote “invalid”.

Reply:
no worries! pushed all commits for review